server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamOptions.java

```diff
@@ -93,4 +101,32 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public static DataStreamOptions fromXContent(XContentParser parser) throws IOException {
         return PARSER.parse(parser, null);
     }
+
+    /**
+     * Creates a class that composes the different fields of the data stream options and normalises explicitly nullified fields.
+     */
+    public static Composer composer(DataStreamOptions options) {
+        return new Composer(options);
+    }
+
+    /**
+     * Composes different data stream options to a normalised final value.
+     */
+    public static class Composer {
```

Comment:
Any reason we went with "Composer" instead of "Builder"? Maybe to distinguish the behavior between the two?

Reply:
Yes, that is correct. I had it as a builder in the beginning, but usually a builder sets fields, while here we will be merging the new value with an existing one. If you think it's too far fetched, I can also name it builder and then in the javadoc mention this distinction. I do not fill strongly about this.